.github/workflows/deploy-docs.yml

```
@@ -3,11 +3,31 @@ name: Deploy documentation website
 on:
   # Runs on pushes targeting the `main` branch
   push:
-    branches: [main]
     paths:
       - "docs/**"
-      - ".github/workflows/deploy-docs.yml" # define the concrete paths on which a change triggers this action, e.g. backend/**
+      - ".github/workflows/deploy-docs.yml" # define the concrete paths on which a change triggers this action, e.g. docs/**
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: pages
+  cancel-in-progress: false
 
 jobs:
-  deploy-pages:
-    uses: it-at-m/.github/.github/workflows/deploy-pages.yml@main
+  deploy-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - id: build_docs
+        uses: it-at-m/.github/.github/actions/action-build-docs@main
+        with:
+          app-path: ${{inputs.app-path}}
+      - id: deploy_docs
+        if: (github.ref_name == 'main')
```

the name of the deploy branch should be configurable (with default to 'main')

---

wenn du weißt wie das geht, kannst du es gerne machen.

ich habe es nicht hinbekommen. ich finde leider jetzt nicht, die Stelle aus dem alten MR. 


Aber hier die Doku dazu https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#context-availability

---

Bei GitHub Pages kann man entweder ein Deployment per Branch oder Action einstellen. Beides geht nicht. Das lässt sich in den Repository-Einstellungen konfigurieren. Wenn man Branch wählt, wird eine Codeänderung auf diesem Branch immer automatisch deployed.

---

```suggestion
        if: (github.ref_name == '${{ inputs.branch-name }} ')
```

Geht das nicht?

bei den inputs müsste dann noch 

```
...
  inputs:
     branch-name:
        type: string
        description: "Branch to trigger deployment to GitHub Pages"
        required: true
        default: main
```

---

@devtobi ich glaube, wovon du redest ist etwas anderes. Da geht es darum, wie deployt wird.

Die @ejcsid meinte man kann "main" noch in eine Input Variable auslagern.

Meiner Meinung nach geht es nicht, weil in `jobs.<job_id>.if` nur die Kontexte von den Variablen `github, needs, vars, inputs` zur Verfügung stehen. 

Aber vielleicht lässt sich das doch auslagern mit dem Kontext `inputs` in einem Action Template. Ich würde es, aber so hardcodiert drin lassen, dann können es die Projekte einfacher anpassen. Zum Schluss gibt es wieder ein Projekt, was nur in Pull-Requests das Ausführen wollen und ein anderes Projekt, was auf zwei Branches etwas ausrollt. Für die ist es dann einfacher hier zu konfigurieren.

---

https://github.com/it-at-m/.github/pull/119 dann lagern wir alles hier raus